composeml/label_maker.py

```diff
@@ -101,21 +102,31 @@ def df_to_labels(df):
                     labels[cutoff_time] = label
 
                 cutoff_time = offset_time(df.index, gap)
+                not_done = progress_bar.n < progress_bar.total
+                progress_bar.update(n=int(not_done))
 
             labels.index = labels.index.rename('cutoff_time')
             labels.index = labels.index.astype('datetime64[ns]')
             return labels.to_frame()
 
-        if verbose:
-            bar_format = "Elapsed: {elapsed} | Remaining: {remaining} | "
-            bar_format += "Progress: {l_bar}{bar}| "
-            bar_format += self.target_entity + ": {n}/{total} "
-            tqdm.pandas(bar_format=bar_format, ncols=90)
-
         df = self._preprocess(df)
         labels = df.groupby(self.target_entity)
-        apply = labels.progress_apply if verbose else labels.apply
-        labels = apply(df_to_labels, *args, **kwargs)
+
+        bar_format = "Elapsed: {elapsed} | Remaining: {remaining} | "
+        bar_format += "Progress: {l_bar}{bar}| "
+        bar_format += self.target_entity + ": {n}/{total} "
+        total = labels.ngroups * num_examples_per_instance
+
+        progress_bar = tqdm(
+            total=total,
+            bar_format=bar_format,
+            disable=not verbose,
+            file=sys.stdout,
+        )
+
+        labels = labels.apply(df_to_labels, progress_bar=progress_bar)
+        n = progress_bar.total - progress_bar.n
```

Comment:
why is this needed? shouldn't it be at total already?

Reply:
If the number of instances that the users sets is more than the number that actually exist in a group, the progress bar will not render completely.
```
labels = label_maker.search(
    df,
    minimum_data="1h",
    num_examples_per_instance=25,
    gap='30d',
    verbose=True,
)
```
```
Elapsed: 00:00 | Remaining: 00:00 | Progress:  14%|██████▌                                       | customer_id: 18/125 
```

Reply:
we should account for that when you break early

```
if df.empty:
    progress_bar.update(n=skipped_iterations)
    break
```

Reply:
Made update to [account for skipped iterations](https://github.com/FeatureLabs/compose-ml/pull/35/files#diff-22bf41f653d21f0ca3dd15dadf5e01e9R95).